src/Acuminator/Acuminator.Analyzers/Analyzers/Localization/LocalizationMessageHelper.cs

```diff
@@ -0,0 +1,136 @@
+                    _syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1052_IncorrectStringToFormat, _messageExpression.GetLocation()));
+                }
+            }
+            else if (IsStringConcatenation())
+            {
+                _syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1053_ConcatinationPriorLocalization, _messageExpression.GetLocation()));
+            }
+        }
+
+        private ITypeSymbol ReadMessageInfo()
+        {
+            if (Cancellation.IsCancellationRequested)
+                return null;
+
+            if (!(_messageExpression is MemberAccessExpressionSyntax memberAccess) ||
+                !(memberAccess.Name is IdentifierNameSyntax messageMember))
+                return null;
+
+            ITypeSymbol messageClassType = memberAccess.Expression
+                                           .DescendantNodesAndSelf()
+                                           .OfType<IdentifierNameSyntax>()
+                                           .Select(i => SemanticModel.GetTypeInfo(i, Cancellation).Type)
+                                           .Where(t => t != null && t.IsReferenceType)
+                                           .FirstOrDefault();
+            if (messageClassType == null)
+                return null;
+
+            ISymbol messageMemberInfo = SemanticModel.GetSymbolInfo(messageMember, Cancellation).Symbol;
+            if (messageMemberInfo == null || messageMemberInfo.Kind != SymbolKind.Field || !messageMemberInfo.IsStatic ||
+                messageMemberInfo.DeclaredAccessibility != Accessibility.Public)
+                return null;
+
+            _messageMember = messageMemberInfo;
+            return messageClassType;
+        }
+
+        private bool IsNonLocalizableMessageType(ITypeSymbol messageType)
+        {
+            if (Cancellation.IsCancellationRequested)
+                return false;
+
+            ImmutableArray<AttributeData> attributes = messageType.GetAttributes();
+            return attributes.All(a => !a.AttributeClass.Equals(Localization.PXLocalizableAttribute));
+        }
+
+        private bool IsIncorrectStringToFormat()
+        {
+            if (!_isFormatMethod || Cancellation.IsCancellationRequested)
+                return false;
+
+            Optional<object> constString = SemanticModel.GetConstantValue(_messageExpression);
+            if (!constString.HasValue)
+                return false;
+
+            Regex regex = new Regex(_formatRegExp, RegexOptions.CultureInvariant);
+            MatchCollection matchesValue = regex.Matches(constString.Value as string);
+
+            return matchesValue.Count == 0;
+        }
+
+        private bool IsStringConcatenation()
+        {
+            if (Cancellation.IsCancellationRequested)
+                return false;
+
+            bool isStringAddition = _messageExpression is BinaryExpressionSyntax && _messageExpression.IsKind(SyntaxKind.AddExpression);
```

Comment:
@metallmonster the AddAssignmentExpression is not considered:
`str += "smthg";`

Reply:
@metallmonster I have thought about this case a little more. The use of `+=` requires a variable and in current implementation we don't support scenario when variable is passed to the localization methods. So, if you decide to leave complex scenarios for later, you don't need to worry about it.

Reply:
@SENya1990 I agree